### PR TITLE
Enabled Docker Image to be built.

### PR DESCRIPTION
**Pull request by gregcorbett**
This PR will allow a APEL 1.6.0 (and future tagged versions with RPMs) Cloud Accounting Docker Image to be built. The primary use case at the moment would be the Indigo DataCloud project, using a separate image for the APEL Server and APEL REST interface to allow for separate scaling of the REST interface.

It adds files needed in every instance of the image, added under `scripts/` and a Dockerfile to do the build.

cron jobs needed to run the summariser should be mounted into `/etc/cron.d/`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+FROM centos:6
```
Comment (tofu-rocketry) [inline Dockerfile]: Could do with a short readme at the top of the file to explain what this file is for and why it's here.

File: scripts/apeldbloader-cloud [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+#
+# Startup script for cloud APEL cloud
```
Comment (tofu-rocketry) [inline scripts/apeldbloader-cloud]: "cloud APEL cloud"?

Could also do with bit more of a readme including an explanation of why this is similar to `dbld`.

File: scripts/apeldbloader-cloud [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#
+# chkconfig: 2345 95 05
+# description: APEL dbloader daemon for cloud
```
Comment (tofu-rocketry) [inline scripts/apeldbloader-cloud]: This isn't the daemon itself - just an init script for the daemon. See the comment for `dbld`.

File: docker/run_on_entry.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
```
Comment (tofu-rocketry) [inline docker/run_on_entry.sh]: Needs readme.

File: scripts/run_cloud_summariser.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
```
Comment (tofu-rocketry) [inline scripts/run_cloud_summariser.sh]: Needs readme.

**Comment by gregcorbett** [comment]
I've added the documentation.

`scripts/apeldbloader-cloud` and `scripts/run_cloud_summariser.sh` could in theory be added to a Server RPM, but don't need to be.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,82 @@
+###########################################################
+# This Dockerfile allows for the creation of a APEL Server
```
Comment (tofu-rocketry) [inline Dockerfile]: "an APEL Server"

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,82 @@
+###########################################################
+# This Dockerfile allows for the creation of a APEL Server 
+# Docker Image for with support for Cloud Accounting
```
Comment (tofu-rocketry) [inline Dockerfile]: Drop "for".

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+#
+# A deployed docker instance will require an instance of
```
Comment (tofu-rocketry) [inline Dockerfile]: Capitalise "Docker".

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+# the APEL Cloud Database and the mounting of
```
Comment (tofu-rocketry) [inline Dockerfile]: "Cloud Accounting Database" and add colon at end of line.

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,82 @@
+###########################################################
+# This Dockerfile allows for the creation of a APEL Server 
+# Docker Image for with support for Cloud Accounting
+#
+# A deployed docker instance will require an instance of
+# the APEL Cloud Database and the mounting of 
+#
+# - a cron job under /etc/cron.d to run
+#   /usr/bin/run_cloud_summariser.sh (atleast) daily
```
Comment (tofu-rocketry) [inline Dockerfile]: space in "at least"

File: Dockerfile [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+# Docker Image for with support for Cloud Accounting
+#
+# A deployed docker instance will require an instance of
+# the APEL Cloud Database and the mounting of 
+#
+# - a cron job under /etc/cron.d to run
+#   /usr/bin/run_cloud_summariser.sh (atleast) daily 
+# - APEL Cloud configuration files under /etc/apel
+#
+# and the optional mounting of
```
Comment (tofu-rocketry) [inline Dockerfile]: Add colon at end.

File: Dockerfile [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+#
+# - a host direcotry under /var/spool/apel/cloud
+#   to persistantly store messages
+# - a mysql configuraiton file under /etc/mysql
```
Comment (tofu-rocketry) [inline Dockerfile]: Capitalisation of "a MySQL"

File: Dockerfile [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+#   (to allow command line access to the MySQL database
+#   from within the container.)
+#
+# It has been placed in the repository root to allow for future
+# docker images to built 'commit by commit' form the exact
```
Comment (tofu-rocketry) [inline Dockerfile]: "from the exact"

File: Dockerfile [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+# state of the source in the repository. At the moment, it is 
+# limited to tagged versions with RPMs built, as that is the
```
Comment (tofu-rocketry) [inline Dockerfile]: "versions that have RPMs"

File: scripts/README.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
+# Scripts included for the Docker image
+
+The following files are present in this directory for the purposes of the Docker image, as they are common to all instances of the APEL cloud server.
+
```
Comment (tofu-rocketry) [inline scripts/README.md]: Put the filenames in code \`backticks\`.

File: scripts/README.md [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+* run_cloud_summariser.sh 
+* apeldbloader-cloud
+
+They could also be included in Server RPMs in the future, so have been included here (rather than the docker directory, which is reserved for files only relevant to the Docker image)
```
Comment (tofu-rocketry) [inline scripts/README.md]: To clarify: "...in the future, but are not at the moment, so..."

Full stop at end.

**Comment by gregcorbett** [comment]
changes made

File: scripts/README.md [hunk 17]
```diff
@@ -0,0 +1,17 @@
+# Scripts included for the Docker image
+
+The following files are present in this directory for the purposes of the Docker image, as they are common to all instances of the APEL cloud server.
+
+* `run_cloud_summariser.sh` 
+* `apeldbloader-cloud`
+
+They could also be included in Server RPMs in the future, but are not at the moment, so have been included here (rather than the docker directory, which is reserved for files only relevant to the Docker image)
```
Comment (tofu-rocketry) [inline scripts/README.md]: Full-stop after parenthesised sentence.

File: scripts/README.md [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+
+## run_cloud_summariser.sh
+
+This script runs the summariser in a safe way, by switching off the cloud loader to ensure there are no writes to the database while summarising.
+## apeldbloader-cloud
```
Comment (tofu-rocketry) [inline scripts/README.md]: Could do with a newline before this section (even though one's added automatically once rendered).

**Comment by gregcorbett** [comment]
changes made, although I haven't rebased yet to preserve the conversation order.

**Comment by gregcorbett** [comment]
This PR has been merged to the Indigo Fork of APEL (see [PR#1](https://github.com/indigo-dc/apel/pull/1)) as they are the primary use case at the moment for a Dockerized APEL server.

As a result, this PR will be closed.

**Pull request closed without merging**